spec/controllers/companies_controller_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+require_relative 'controller_shared_examples'
+
+
+RSpec.describe CompaniesController, type: :controller do
+
+  describe '#index will fix_FB_changed_params' do
+
+    it "does not change URL if there are no query ('q') parameters" do
+      no_query_params = { "utf8" => "✓" }
+
+      expected_fixed = { "utf8" => "✓",
+                         "controller" => "companies",
+                         "action" => "index",
+      }
+
+      get :index, params: no_query_params
+
+      expect(subject.params.to_unsafe_h).to eq expected_fixed
+
+    end
+
+    it 'q parameters that are a Hash are converted to Array with the Hash values' do
+
+      fb_mangled_params = { "utf8" => "✓",
+                            "q" => {
+                                "business_categories_id_in" => { "0" => "2", "1" => "2", "2" => "2" }
+                            }
+      }
+
+      expected_fixed_q = { "business_categories_id_in" => ["2", "2", "2"] }
+
+      get :index, params: fb_mangled_params
+      expect(subject.params.to_unsafe_h['q']).to eq expected_fixed_q
+    end
+
+    it 'empty values do not need to be retained)' do
+
+      fb_mangled_params = { "utf8" => "✓",
+                            "q" => {
+                                "business_categories_id_in" => { "0" => "2", "1" => "2", "2" => "2" },
+                                "addresses_region_id_in" => { "0" => "6" },
+                                "addresses_kommun_id_in" => { "0" => "" },
+                                "name_in" => { "0" => "" } },
+                            "commit" => "Sök" }
```

Comment:
I think the pattern we've seen is like this .... 

What we would see without mangling:
```
Parameters: {"utf8"=>"✓", "q"=>{"business_categories_id_in"=>["", "2"], "addresses_region_id_in"=>["", "6"], "addresses_kommun_id_in"=>[""], "name_in"=>[""]}, "commit"=>"Sök"}
```
versus the mangled version:
```
Parameters: {"utf8"=>"✓", "q"=>{"business_categories_id_in"=>{"0"=>nil, "1"=>"2"}, "addresses_region_id_in"=>{"0"=>nil, "1"=>"6"}, "addresses_kommun_id_in"=>{"0"=>nil}, "name_in"=>{"0"=>nil}}, "commit"=>"Sök"}
```
That is, we don't see the occurrence of `{ "0" => ""  }` but rather `{ "0"=>nil }`.

So perhaps a more realistic version of a mangled test hash is:
```
{ "utf8" => "✓",
                            "q" => {
                                "business_categories_id_in" => { "0" => "1", "1" => "2", "2" => "3" },
                                "addresses_region_id_in" => { "0" => "6" },
                                "addresses_kommun_id_in" => { "0" => nil },
                                "name_in" => { "0" => nil } },
                            "commit" => "Sök" }
```
(note that I also changed the cat ID's to be three different values, since we'd never really see the same value 3 times).